src/pages/index.js

```diff
@@ -0,0 +1,282 @@
+  const cardImageEL = cardElement.querySelector(".card__image");
+  const cardLikeBtn = cardElement.querySelector(".card__like-btn");
+  const cardDeleteBtn = cardElement.querySelector(".card__delete-btn");
+  cardNameEL.textContent = data.name;
+  cardImageEL.src = data.link;
+  cardImageEL.alt = data.name;
+
+  cardImageEL.addEventListener("click", () => {
+    openModal(previewModal);
+    previewModalCaptionEL.textContent = data.name;
+    previewModalImageEL.src = data.link;
+    previewModalImageEL.alt = data.name;
+  });
+  cardLikeBtn.addEventListener("click", () => {
+    cardLikeBtn.classList.toggle("card__like-btn_liked");
+  });
+  cardDeleteBtn.addEventListener("click", (evt) => {
+    handleDeleteCard(cardElement, data._id);
+  });
+
+  return cardElement;
+}
+cardModalClosedBtn.addEventListener("click", () => {
+  closeModal(cardModal);
+});
+const previewModalClosedBtn = previewModal.querySelector(".modal__close-btn");
+previewModalClosedBtn.addEventListener("click", () => {
+  closeModal(previewModal);
+});
+const modals = document.querySelectorAll(".modal");
+modals.forEach((modal) =>
+  modal.addEventListener("mousedown", handleOverlayClick)
+);
+function handleEscapeKey(evt) {
+  if (evt.key === "Escape") {
+    const openedModal = document.querySelector(".modal_opened");
+    closeModal(openedModal);
+  }
+}
+function handleOverlayClick(evt) {
+  if (evt.target.classList.contains("modal_opened")) {
+    closeModal(evt.target);
+  }
+}
+
+function handleLike(evt, id) {
+  const likeButton = evt.target;
+  const isLiked = likeButton.classList.contains("card__like-btn_active");
+  api
+    .changeLikeCardStatus(id, !isLiked)
+    .then((updatedCard) => {
+      // Update the UI accordingly
+      likeButton.classList.toggle("card__like-btn_active", !isLiked);
+      const likeCounter = likeButton
+        .closest(".card")
+        .querySelector(".card__like-count");
+      if (likeCounter) {
+        likeCounter.textContent = updatedCard.likes.length;
+      }
+    })
+    .catch((err) => {
+      console.error("Error updating like status:", err);
+    });
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscapeKey);
+}
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscapeKey);
+}
+function handleEditFormSubmit(evt) {
+  const submitBtn = evt.submitter;
+  submitBtn.textContent = "Saving...";
+  setButtontext(submitBtn, false, "Saving...");
+  evt.preventDefault();
+  api
+    .editUserInfo({ name: editModalNameInput.value, about: editDescriptionInput.value, })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      // submitBtn.textContent = "Save";
+      setButtontext(submitBtn, true);
```

Comment:
Please, use `setButtontext` correctly for changing the button text in `.finally` as well


```js

    .finally(() => {
      setButtontext(submitBtn, false); // this will show `Save`
    });

```


You should use `false`, to return the button text back